Ghost protocol upgrade files/migration-extraction-guide.sh

```
@@ -0,0 +1,368 @@
+# Migration Guide: Extract Memory System to Standalone Service
+# From sd-ghost-protocol to Memory-as-a-Service
+
+## Phase 1: Prepare the Extraction
+
+# 1. Create new service directory structure
+mkdir memory-service
+cd memory-service
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)